lib/static/modules/static-image-accepter.ts

```diff
@@ -0,0 +1,134 @@
+    commitedImages: Record<string, string>
+    }
+
+type CommitPayload = Array<{
+    id: string;
+    stateNameImageId: string;
+    image: string;
+    path: string;
+}>
+
+const STATIC_ACCEPTER_PERSISTENT_LOCAL_STORAGE_KEY = 'static-image-accepter';
+
+export const checkIsEnabled = (config: ReporterConfig['staticImageAccepter'], isGui: boolean): boolean => {
+    if (isEnabled !== null) {
+        return isEnabled;
+    }
+
+    if (!config || !config.enabled || isGui) {
+        return isEnabled = false;
+    }
+
+    const requiredConfigFields: Array<keyof ReporterConfig['staticImageAccepter']> = ['repositoryUrl', 'pullRequestUrl', 'serviceUrl'];
+    const missingFields = (requiredConfigFields).filter(key => !config[key]);
+
+    if (missingFields.length) {
+        console.error(`staticImageAccepter is disabled. Invalid config: '${missingFields.join(', ')}' is(/are) missing!`);
+        return isEnabled = false;
+    }
+
+    return isEnabled = true;
+};
+
+export const formatCommitPayload = (
+    acceptableImgages: AcceptableImage[],
+    imagesById: ImagesById,
+    extraImages: Array<{imageId: string, stateNameImageId: string}> = []
+): CommitPayload => {
+    const imagesToCommit = acceptableImgages
+        .filter(image => image.commitStatus === STAGED)
+        .map(image => ({imageId: image.id, stateNameImageId: image.stateNameImageId}))
+        .concat(extraImages);
+
+    if (imagesToCommit.find(({imageId}) => !imagesById[imageId].refImg.relativePath)) {
+        throw new Error(`The version of your tool does not support static image accepter: missing "relativePath"`);
+    }
+
+    return imagesToCommit.map(({imageId, stateNameImageId}) => ({
+        id: imageId,
+        stateNameImageId,
+        image: imagesById[imageId].actualImg.path,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        path: imagesById[imageId].refImg.relativePath!
+    }));
+};
+
+const getLocalStorageCommitedImages = (): Record<string, string> => {
+    const date = get(window, ['data', 'date']);
+    const pathName = get(window, ['location', 'pathname']);
+
+    const value = localStorage.getItem(STATIC_ACCEPTER_PERSISTENT_LOCAL_STORAGE_KEY, null) as LocalStorageValue;
+
+    if (!value || value.date !== date || value.pathName !== pathName) {
+        return {};
+    }
+
+    return value.commitedImages;
+};
+
+export const getLocalStorageCommitedImageIds = (): string[] => Object.values(getLocalStorageCommitedImages());
+
+export const storeCommitInLocalStorage = (newCommitedImages: Array<{id: string, stateNameImageId: string}>): void => {
+    const date = get(window, ['data', 'date']);
+    const pathName = get(window, ['location', 'pathname']);
+
+    const currentState = getLocalStorageCommitedImages();
+    const update = newCommitedImages.reduce((acc, newCommitedImage) => {
+        acc[newCommitedImage.stateNameImageId] = newCommitedImage.id;
+
+        return acc;
+    }, {} as Record<string, string>);
```

Comment:
Saving images with `stateNameImageId` key to overwrite existing commited images with the same `stateNameImageId` while keeping the other ones